pkg/backends/otlp/internal/data/map.go

```diff
@@ -70,22 +79,40 @@ func (m Map) Insert(key string, value string) {
 			Value: val,
 		})
 	default:
-		if v, ok := (*m.raw)[index].Value.Value.(*v1common.AnyValue_StringValue); ok && v.StringValue == value {
-			return
-		}
-
-		v, ok := (*m.raw)[index].Value.Value.(*v1common.AnyValue_ArrayValue)
-		if !ok {
-			// Convert a simple key value into an array of values
-			v = &v1common.AnyValue_ArrayValue{
+		v := (*m.raw)[index].Value
+		switch t := v.Value.(type) {
+		case *v1common.AnyValue_StringValue:
+			if t.StringValue == value {
+				return
+			}
+			insert := &v1common.AnyValue_StringValue{
+				StringValue: value,
+			}
+			v.Value = &v1common.AnyValue_ArrayValue{
 				ArrayValue: &v1common.ArrayValue{
 					Values: []*v1common.AnyValue{
-						{Value: v},
+						{Value: t},
+						{Value: insert},
 					},
 				},
 			}
+		case *v1common.AnyValue_ArrayValue:
+			exist := slices.ContainsFunc(t.ArrayValue.Values, func(av *v1common.AnyValue) bool {
+				return av.GetStringValue() == value
+			})
+			if exist {
+				return
+			}
+			t.ArrayValue.Values = append(t.ArrayValue.Values, &v1common.AnyValue{
+				Value: &v1common.AnyValue_StringValue{
+					StringValue: value,
+				},
+			})
 		}
-		v.ArrayValue.Values = append(v.ArrayValue.Values, val)
+		slices.Insert((*m.raw), index, &v1common.KeyValue{
```

Comment:
I think we're inserting even if the value is appended to or set, I think this will lead to deplicate resources